tket/src/Circuit/Boxes.cpp

```diff
@@ -285,15 +285,35 @@ bool CompositeGateDef::operator==(const CompositeGateDef &other) const {
   return this->get_def()->circuit_equality(*other.get_def(), {}, false);
 }
 
+static const composite_def_ptr_t &get_non_null_ptr(
```

Comment:
It could be. Currently the `CustomGate` constructor calls the `Box` constructor, which requires a `signature` coming from the `gate->signature()` function.

Thus we'd have to call the `Box` constructor without a  `signature` object, THEN do the non-null check on `gate` and set the `op_signature_t signature_` to `gate->signature()` (which is just a vector) within the body of the `CustomGate` constructor.

This might slow it down (although negligibly), because a useless empty `signature` vector is constructed to pass into the `Box` constructor (although, if we're lucky, the compiler would optimise it away).

However I note that `Box` is already effectively doing that because `signature = {}` is a default argument `Box` constructor, so already a useless empty `signature` vector must be constructed (unless the compiler optimises it away).

So, I'll do that if you're happy with the possible performance hit.

Reply:
I think that's fine.

Also note that you will need to do a merge from `develop` to get the Linux tests to pass.